geoh5py/objects/geo_image.py

```diff
@@ -552,3 +525,80 @@ def to_grid2d(
         :return: the new created :obj:`geoh5py.objects.grid2d.Grid2D`.
         """
         return self.converter.to_grid2d(self, transform, **grid2d_kwargs)
+
+    @property
+    def rotation(self) -> float | None:
+        """
+        The rotation of the image in degrees
+        :return: the rotation angle.
+        """
+        if self._rotation is None and self._vertices is not None:
+            # Get the x and y values of the first and second corners
+            corner_1_x = self._vertices["x"][0]
+            corner_1_y = self._vertices["y"][0]
+            corner_2_x = self._vertices["x"][1]
+            corner_2_y = self._vertices["y"][1]
+
+            # Form them into numpy arrays
+            corner_1 = np.array([corner_1_x, corner_1_y])
+            corner_2 = np.array([corner_2_x, corner_2_y])
+
+            # Get the vector from corner_1 to corner_2
+            delta = corner_2 - corner_1
+
+            # Reference direction (the x-axis)
+            reference = np.array([1, 0])
+
+            # Calculate the angle between the reference direction and the vector
+            rotation_rad = np.arctan2(
+                np.cross(reference, delta), np.dot(reference, delta)
+            )
+
+            # Convert to degrees
+            rotation_deg = np.rad2deg(rotation_rad)
+
+            # Adjust the angle to lie between [0, 360)
+            if rotation_deg < 0:
+                rotation_deg = 360 + rotation_deg
+
```

Comment:
Could potentially simplify to

```suggestion
        dx, dy = np.diff(self._vertices["x"][:2]), np.diff(self._vertices["y"][:2])
        rotation_rad = np.arctan2(dy, dx)
        self._rotation = np.rad2deg(rotation_rad)
```

Reply:
You should double-check the suggestion, but feels like it should work.